validator/src/micro.rs

```diff
@@ -9,21 +9,18 @@ use std::{
 use futures::{future::BoxFuture, ready, FutureExt, Stream};
 use nimiq_block::{Block, EquivocationProof, MicroBlock, SkipBlockInfo};
 use nimiq_blockchain::{BlockProducer, BlockProducerError, Blockchain};
-use nimiq_blockchain_interface::{AbstractBlockchain, PushResult};
+use nimiq_blockchain_interface::AbstractBlockchain;
 use nimiq_mempool::mempool::Mempool;
 use nimiq_time::sleep;
 use nimiq_utils::time::systemtime_to_timestamp;
 use nimiq_validator_network::ValidatorNetwork;
 use nimiq_vrf::VrfSeed;
 use parking_lot::RwLock;
 
-use crate::aggregation::skip_block::SkipBlockAggregation;
+use crate::{aggregation::skip_block::SkipBlockAggregation, validator::BlockPublisher};
 
-// Ignoring this clippy warning since size difference is not that much (320
-// bytes) and we probably don't want the performance penalty of the allocation.
-#[allow(clippy::large_enum_variant)]
 pub(crate) enum ProduceMicroBlockEvent {
-    MicroBlock(MicroBlock, PushResult),
+    MicroBlock,
 }
```

Comment:
Yes it is unused and could be removed.

Removing it will likely also entail making the micro Producer no longer having to implement Stream. That probably will mean a bunch of additional changes (for the better).

I opted not to do it here as it is out of scope of this PR.

Reply:
Maybe the "fix clippy warning" commit could be dropped instead of half-removing the `MicroBlock` event. A `#[allow()]` could be used instead, leaving the (full) removal for a future PR.

Reply:
For the time being I don't plan on doing any additional work there. The Event is unused, but removing the stream makes us less flexible in the future. If the micro producer is ever going to return something meaningful again it would have to be re implemented. It does not break anything as it is and I think it is semantically not too bad either.